pkg/cloud/azure/services/certificates/certificates.go

```diff
@@ -40,24 +42,25 @@ import (
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/cmd/kubeadm/app/util/pubkeypin"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services"
 )
 
 // Get should implement returning certs and kubeconfigs.
-func (s *Service) Get(ctx context.Context) (interface{}, error) {
+func (s *Service) Get(ctx context.Context, spec interface{}) (interface{}, error) {
 	return nil, errors.New("Not implemented")
 }
 
 // Delete cleans up and generated certificates, could be useful for renewal.
-func (s *Service) Delete(ctx context.Context) error {
+func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	return nil
 }
 
 // CreateOrUpdate Helper function so this can be unittested.
-func (s *Service) CreateOrUpdate(ctx context.Context) error {
+func (s *Service) CreateOrUpdate(ctx context.Context, spec interface{}) error {
+	klog.V(2).Infof("generating certificates")
 	clusterName := s.scope.Cluster.Name
 	tmpDirName := "/tmp/cluster-api/" + clusterName
```

Comment:
Consider extracting this to remove the dependency on the path, and on the concrete file system.
Relying on io.Reader makes it easy to test without touching the filesystem

Reply:
Agreed, side effect of using kubeadm which always writes to filesystem, not sure if there are internal API to achieve that, please file an issue and will be addressed in future PR.

Reply:
Let's punt to another PR. @serbrech -- please file an issue for this.